### DISCUSSION TITLE
Is it possible to define two layouts for different groups in stencil container?

### DISCUSSION THREAD

**Discussion by gintarasgraciovas** [clientIO/joint#2568] [Q&A] [answered]
### Introduction

I have a few groups with assets in stencil container. The problem is that our assets have different sizes and I have to specify different layouts as well. Is there any ways to do so without creating additional stencil or with creation of a new stencil, but in the same place or smth...?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
Yes. The Stencil `layout` option can be:
- defined as a function
- you can override the `layout` options on per group-basis

Here's an example of the latter: https://www.jointjs.com/demos/planogram
_(Note that drinks are spread over 4 columns)_

**Reply by gintarasgraciovas** [reply]
Thanks a lot for the quick response. I'll check this out! :)

**Reply by gintarasgraciovas** [reply]
It's working perfectly! Just one remaining question... Is it possible to use something like flex display in such a way that if there are two columns, but one item is too large, it can span across both columns?

**Reply by kumilingus** [reply]
The default layout used in the Stencil is `GridLayout` which does not support this.
But the `layout` option could be defined as a function so you can run a custom layout. For instance a [Serpentine]( https://codepen.io/jointjs/pen/eYjXjrb) layout.
